cookbook/bundles/extension.rst

```diff
@@ -125,3 +125,42 @@ Using Configuration to Change the Services
 The Extension is also the class that handles the configuration for that
 particular bundle (e.g. the configuration in ``app/config/config.yml``). To
 read more about it, see the ":doc:`/cookbook/bundles/configuration`" article.
+
+Adding Classes to Compile
+-------------------------
+
+In order to make applications run as fast as possible on production environment,
+Symfony creates a big ``classes.php`` file in the cache directory. This file
+aggregates the contents of the PHP classes that are used in every request,
+reducing the I/O operations related to those classes.
+
+Your own bundles can add classes into this file thanks to the ``addClassesToCompile()``
+method. Define the classes to compile as an array of their fully qualified class
+names::
+
+    // ...
+    public function load(array $configs, ContainerBuilder $container)
+    {
+        // ...
+
+         $this->addClassesToCompile(array(
+            'AppBundle\\Manager\\UserManager',
+            'AppBundle\\Service\\Slugger',
+            // ...
+        ));
+    }
+
+.. note::
+
+    If some class extends from other classes, all its parents are included
+    automatically in the list of classes to compile.
+
+After adding to compile all the classes commonly used by your bundle, you can
+expect a minor performance improvement.
+
+The main drawback of this technique is that it doesn't work when:
+
+* Classes contain annotations, such as controllers with ``@Route`` annotations
+  and entities with ``@ORM`` or ``@Assert`` annotations;
+* Classes use the ``__DIR__`` and ``__FILE__`` constants, because their values
+  will change when loading these classes from the ``classes.php`` file.
```

Comment:
I personally would find it interesting to know why it doesn't work for annotations. One of the reasons is the file location retrieved from reflection. Is this something you could add to the documentation?